sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -901,7 +901,8 @@ class SQLContext private[sql](
   @transient
   protected[sql] val prepareForExecution = new RuleExecutor[SparkPlan] {
     val batches = Seq(
-      Batch("Add exchange", Once, EnsureRequirements(self))
+      Batch("Add exchange", Once, EnsureRequirements(self)),
+      Batch("Replace PhysicalRelation", FixedPoint(100), ReplacePhysicalRelation(self))
```

Comment:
Does this need to be run more than once?

"Replace PhysicalRelation" sounds too general